shell/browser/mac/in_app_purchase_product.mm

```diff
@@ -156,9 +216,64 @@ - (NSString*)formatPrice:(NSDecimalNumber*)price
       }
     }
   }
+  if (@available(macOS 10.13.2, *)) {
+    if (product.introductoryPrice != nil) {
+      productStruct.introductoryPrice =
+          [self skProductDiscountToStruct:product.introductoryPrice];
+    }
+    if (product.subscriptionPeriod != nil) {
+      productStruct.subscriptionPeriod =
+          [self skProductSubscriptionPeriodToStruct:product.subscriptionPeriod];
+    }
+  }
+  if (@available(macOS 10.14, *)) {
+    if (product.subscriptionGroupIdentifier != nil) {
+      productStruct.subscriptionGroupIdentifier =
+          [product.subscriptionGroupIdentifier UTF8String];
+    }
+  }
+  if (@available(macOS 10.14.4, *)) {
+    if (product.discounts != nil) {
+      productStruct.discounts.reserve([product.discounts count]);
+
+      for (SKProductDiscount* discount in product.discounts) {
+        productStruct.discounts.push_back(
+            [self skProductDiscountToStruct:discount]);
+      }
+    }
+  }
 
   // Downloadable Content Information
   productStruct.isDownloadable = [product downloadable];
+  if (@available(macOS 10.14, *)) {
+    if (product.downloadContentVersion != nil) {
+      productStruct.downloadContentVersion =
+          [product.downloadContentVersion UTF8String];
+    }
+    if (product.downloadContentLengths != nil) {
+      productStruct.downloadContentLengths.reserve(
+          [product.downloadContentLengths count]);
+
+      for (NSNumber* contentLength in product.downloadContentLengths) {
+        productStruct.downloadContentLengths.push_back(
+            [contentLength longLongValue]);
+      }
+    }
+  } else {
+    if (product.contentVersion != nil) {
+      productStruct.downloadContentVersion =
+          [product.contentVersion UTF8String];
+    }
+    if (product.contentLengths != nil) {
+      productStruct.downloadContentLengths.reserve(
+          [product.contentLengths count]);
+
+      for (NSNumber* contentLength in product.contentLengths) {
+        productStruct.downloadContentLengths.push_back(
+            [contentLength longLongValue]);
+      }
+    }
+  }
```

Comment:
`downloadContentVersion` and `downloadContentLengths` are intended to replace `contentVersion` and `contentLengths` respectively. http://codeworkshop.net/objc-diff/sdkdiffs/macos/10.14/StoreKit.html